src/controller/CHIPDeviceController.h

```diff
@@ -49,6 +49,30 @@ namespace Controller {
 constexpr uint16_t kNumMaxActiveDevices = 64;
 constexpr uint16_t kNumMaxPairedDevices = 128;
 
+struct ControllerInitParams
+{
+    PersistentStorageDelegate * storageDelegate = nullptr;
+    System::Layer * systemLayer                 = nullptr;
+    Inet::InetLayer * inetLayer                 = nullptr;
+
+    ControllerInitParams() {}
```

Comment:
I realize this pattern is elsewhere, but I don't understand why we need all this.

What's wrong with

```
struct ControllerInitParams
{
    PersistentStorageDelegate * storageDelegate = nullptr;
    System::Layer * systemLayer                 = nullptr;
    Inet::InetLayer * inetLayer                 = nullptr;
};
```

?

(maybe I just don't understand the desire to build a whole object as a one-liner. Personally, I think the lack of line breaks is a negative for readability)

Reply:
I found even if we dont use the setter, we can still write code in one-liner style.
Since this is just a wrapper of several params (and to avoid merge conflicts), I think the bare struct without getter/setter is ok.